src/OpenApi/src/Services/OpenApiDocumentProvider.cs

```diff
@@ -25,10 +25,17 @@ internal sealed class OpenApiDocumentProvider(IServiceProvider serviceProvider)
     /// <param name="writer">A text writer associated with the document to write to.</param>
     public async Task GenerateAsync(string documentName, TextWriter writer)
     {
+        // We need to retrieve the document name in a case-insensitive manner to support case-insensitive document name resolution.
```

Comment:
I think we can just reference the comment in the `IServiceCollection` extensions file instead of duplicating the content here.